src/OrleansRuntime/Catalog/DefaultGrainActivator.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Globalization;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Orleans.Runtime
+{
+    /// <summary>
+    /// <see cref="IGrainActivator"/> that uses type activation to create grains.
+    /// </summary>
+    public class DefaultGrainActivator : IGrainActivator
+    {
+        private readonly Func<Type, ObjectFactory> createFactory = type => ActivatorUtilities.CreateFactory(type, Type.EmptyTypes);
```

Comment:
Concerns:
Insufficient design - Grain lifecycle needs work, I think we all agree, but it's also very important to get it right.  I am of the opinion that upfront design is necessary.  I do not suspect that it is sufficient or desirable to simply apply the mvc controller pattern to this because they look similar.
I advocate that we do the design work necessary to address grain lifecycle needs, then, after due diligence, if the resulting design is comparable to MVC's controller pattern, then we apply it as it is more familiar to users.  This ensures we're applying the right pattern not forcing the solution into a pattern that is already familiar.
Churn - The issue that Eldar is experiencing is in part due to the fact that we've been working on this system piecemeal, creating churn.  I advocate up front design to make fewer and more complete changes with less churn.  Making a minimal change to address the previous braking change is churn, but not as much as performing this change and the risk of further changes due to lack of upfront design.
Risk - We want to get this in for 1.5, which is already in beta, which means this change will not get as much community testing as the rest of 1.5, and this is critical code.  A minimal change is far less risky.

These are my concerns, however I will happily defer to the teams decision if most of those interested in this change do not share my concerns.